src/utils/string.js

```diff
@@ -4,21 +4,23 @@
 
 // @flow
 
-/**
- * Takes a string and returns the string with public URLs removed.
- * It doesn't remove the URLs like `chrome://..` because they are internal URLs
- * and they shouldn't be removed.
- */
-export function removeURLs(
-  string: string,
-  removeExtensions: boolean = true,
-  redactedText: string = '<URL>'
-): string {
-  const protocols = ['http', 'https', 'ftp', 'file', 'moz-page-thumb'];
-  if (removeExtensions) {
-    protocols.push('moz-extension');
-  }
-  const regExp = new RegExp(
+import memoize from 'memoize-immutable';
+
+// Lazily initialize the RegExp for removeURLs at the first time of use.
+// removeURLs is in a hot path during sanitization and the RegExp initialization
+// is taking a lot of time. We are memoizing it here to reduce the cost of this
+// during every single use.
+const _getRemoveURLsRegExp = memoize(() => {
```

Comment:
I think we should just create the regexp statically: this is simpler, works fine, and performs better.

memoize does a lot of things :-) which are usually not a big problem, but in a hot loop like this one and because we have a good and simple alternative, I believe we shouldn't use it.

a memoized function: https://github.com/memoize-immutable/memoize-immutable/blob/f4447e041ce7956455c9276191f5314d2bea08ea/index.js#L29-L36
Tuplemap (the default cache for memoize): https://github.com/memoize-immutable/TupleMap/blob/master/index.js (look especially at `_hash` which is used for get/set/has.

Reply:
Constructing a regexp is pretty cheap -- of course not cheap enough to do it in a hot loop, but cheap enough to do it once at loadtime.
Calling `memoize` itself is also a lot of operations, it's not especially clear this would be cheaper than constructing a regexp.

Moreover I'm not convinced about the use of an array to build this regexp. I think this was useful when adding moz-extension was optional, but now it's all constant, so I believe we can use a plain simple regexp using `/.../`.

Reply:
I like the idea of creating an array for two reasons:
1. It's easier to see what kind of protocols are supported by the RegExp. Otherwise it's more obscured. You need to look at the RegExp carefully to see. Also it simplifies the RegExp a bit. I think it's good since many people are intimidated by it :)
2. It's easier to maintain. You can just add a new protocol to this array, and we join this automatically. You don't have to change the comment and its indentation below or take care of the linter all the time.

Since this is an intermediate array that we create only once, I felt like this is nice to keep. I don't really see a negative side of having this array when it's inside a memoized function. But I'm not sure about having this as a global variable, probably not so good :/

For memoization, I see some similar usages in the codebase, that's why I wanted to use it as well. Like `dateFormats` here:
https://github.com/firefox-devtools/profiler/blob/3d59957dd2592622157fc6fe3cd0ec07ac6a2bd9/src/components/app/ListOfPublishedProfiles.js#L37-L59

But I can still change that to a global initialized regexp if you want. My only concern was that I didn't want to initialize it if we are not going to use it at all. But if we think that the memoize's cost is more than the regexp initialization, then why not.